src/doc-printer.js

```diff
@@ -176,6 +194,51 @@ function printDocToString(doc, width, newLine) {
               break;
           }
           break;
+        case "fill":
+          // shouldRemeasure = false;
+          let rem = width - pos;
+
+          const parts = doc.parts;
+          if (parts.length === 1) {
+            cmds.push([ind, MODE_BREAK, parts[0]]);
+            break;
+          }
+
+          const first = parts[0];
+          const second = parts[1];
+          const remaining = parts.slice(1);
+          const firstCmd = [ind, MODE_FLAT, first];
+          const remainingCmd = [ind, MODE_BREAK, fill(remaining)];
+
+          const firstAndSecondCmd = [ind, MODE_FLAT, concat([first, second])];
+
+          if (fits(firstAndSecondCmd, cmds, rem, true)) {
+            // console.log('first and second fit', [first, second]);
+            cmds.push(remainingCmd)
+            cmds.push([ind, MODE_FLAT, line]);
+            cmds.push(firstCmd);
+          } else if (fits(firstCmd, cmds, width - rem, true)) {
+            // console.log('only first fits', [first, second]);
+            cmds.push(remainingCmd)
+            if (typeof second !== 'string' || typeof first !== 'string') {
+              cmds.push([ind, MODE_BREAK, "{' '}"]);
```

Comment:
Yep, this felt strange but it was the easiest way to get something sane printing out!

Ideally the `fill` implementation would be reasonably agnostic, but I don't know enough to make that work yet.